### PR TITLE
Handle negative timestamps

### LINKED ISSUE #3367: negative timestamps not valid for line protocol
CLI, showing max positive int64 is valid, but the smallest negative integer fails.

```
> insert timestamps value=1 
> insert timestamps value=1 9023372036854775808
> insert timestamps value=1 -1
ERR: unable to parse ' timestamps value=1 -1': bad timestamp

> 
```

`curl` showing same issue (not sure why curl tops out at max int64 -1):

```
$ curl -XPOST 'http://localhost:8086/write?db=mydb' -d 'timestamps value=1 9223372036854775808'
unable to parse 'timestamps value=1 9223372036854775808': strconv.ParseInt: parsing "9223372036854775808": value out of range
$ curl -XPOST 'http://localhost:8086/write?db=mydb' -d 'timestamps value=1 9223372036854775807'
$ curl -XPOST 'http://localhost:8086/write?db=mydb' -d 'timestamps value=1 -1'
unable to parse 'timestamps value=1 -1': bad timestamp
```

### PR DESCRIPTION
**Pull request by gunnaraasen**
Fixes #3367.

### CODE REVIEW

**Comment by otoolep** [comment]
What does a negative timestamp actually mean now?

**Comment by gunnaraasen** [comment]
A negative timestamp is the number of nanoseconds earlier than the epoch. Right now, the line protocol does not support writing timestamps earlier than `0` (i.e. 1970-01-01T00:00:00Z) With this change the earliest timestamp will be `-9223372036854775807` (i.e. 1677-09-21 00:12:43.145224193Z).

Also added a couple tests added to confirm the upper and lower bounds the line protocol will parse (e.g. the max and min int64 values).

**Comment by otoolep** [comment]
I guess I am wondering why we would support this, but I guess if we can code for it without much work, it's OK.

cc @jwilder

**Comment by otoolep** [comment]
As opposed to just rejecting the point.

**Comment by otoolep** [comment]
+1 on the code, not so sure about whether we shouldn't instead kick back an error.

**Comment by gunnaraasen** [comment]
It does double the supported time range with minimal cost.

**Comment by beckettsean** [comment]
@otoolep 0.9 has always supported negative timestamps for writes and reads. This fixes a bug in the line protocol where it threw an error when parsing a negative epoch timestamp. The JSON write path still happily allows negative timestamps and the entire query engine is fine with it.

**Comment by beckettsean** [comment]
When this is merged we need to update the docs

**Comment by otoolep** [comment]
OK, thanks @beckettsean -- sounds like this should be merged.

File: models/points_test.go [hunk 1]
```diff
@@ -975,6 +975,65 @@ func TestParsePointUnicodeString(t *testing.T) {
...
+	}
+	_, err = models.ParsePointsString("cpu value=1 -")
+	if err == nil {
+		t.Fatalf("ParsePoints failed: %v", err)
+	}
+	_, err = models.ParsePointsString("cpu value=1 -/")
+	if err == nil {
+		t.Fatalf("ParsePoints failed: %v", err)
+	}
+	_, err = models.ParsePointsString("cpu value=1 -1?")
```
Comment (jwilder) [inline models/points_test.go]: Maybe add a test of `1-` as well.  Looks like that should pass but good for coverage.

**Comment by jwilder** [comment]
Small test suggestion but :+1: otherwise.

File: models/points_test.go (hunk 1 above)
Comment (gunnaraasen) [inline models/points_test.go]: Good idea

**Comment by gunnaraasen** [comment]
Added the suggested test. Going to merge on green.

**Pull request merged**